src/github.com/cloudfoundry-incubator/webhooks/pkg/webhooks/manager/event.go

```diff
@@ -0,0 +1,182 @@
+		return nil, err
+	}
+	oldCrd, err := getGenericResource(ar.Request.OldObject.Raw)
+	if err != nil {
+		glog.Errorf("Could not get the old GenericResource object %v", err)
+		return nil, err
+	}
+	crd.Status.lastOperation = getLastOperation(crd)
+	crd.Spec.options = getOptions(crd)
+	oldCrd.Status.lastOperation = getLastOperation(oldCrd)
+	oldCrd.Spec.options = getOptions(oldCrd)
+	crd.Status.appliedOptions = getAppliedOptions(crd)
+	oldCrd.Status.appliedOptions = getAppliedOptions(oldCrd)
+	return &Event{
+		AdmissionReview: ar,
+		crd:             crd,
+		oldCrd:          oldCrd,
+	}, nil
+}
+
+func (e *Event) isStateChanged() bool {
+	loNew := e.crd.Status.lastOperation
+	loOld := e.oldCrd.Status.lastOperation
+	glog.Infof("New: type: %s, state: %s\n", loNew.Type, loNew.State)
+	glog.Infof("Old: type: %s, state: %s\n", loOld.Type, loOld.State)
+	return loNew.Type == loOld.Type && loNew.State != loOld.State
+}
+
+func (e *Event) isPlanChanged() bool {
+	appliedOptionsNew := e.crd.Status.appliedOptions
+	appliedOptionsOld := e.oldCrd.Status.appliedOptions
+	return appliedOptionsNew.PlanID != appliedOptionsOld.PlanID
+}
+
+func (e *Event) isCreate() bool {
+	return e.crd.Status.lastOperation.Type == "create"
+}
+
+func (e *Event) isUpdate() bool {
+	return e.crd.Status.lastOperation.Type == "update"
+}
+
+func (e *Event) isSucceeded() bool {
+	return e.crd.Status.lastOperation.State == "succeeded"
+}
+
+func (e *Event) isMeteringEvent() bool {
+	if e.isStateChanged() && e.isSucceeded() {
+		if e.isUpdate() && e.isPlanChanged() {
+			return true
+		}
+		if e.isCreate() {
+			return true
+		}
+	}
+	return false
+}
+
+// ObjectToMapInterface converts an Object to map[string]interface{}
+func ObjectToMapInterface(obj interface{}) (map[string]interface{}, error) {
+	values := make(map[string]interface{})
+	options, err := json.Marshal(obj)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(options, &values)
+	if err != nil {
+		return nil, err
+	}
+	return values, nil
+}
+
+func getClient(cfg *rest.Config) (client.Client, error) {
+	glog.Infof("setting up manager")
+	mgr, err := manager.New(cfg, manager.Options{})
+	if err != nil {
+		glog.Errorf("unable to set up overall controller manager %v", err)
+		return nil, err
+	}
+	options := client.Options{
+		Scheme: mgr.GetScheme(),
+		Mapper: mgr.GetRESTMapper(),
+	}
+	apiserver, err := client.New(cfg, options)
+	if err != nil {
+		glog.Errorf("unable create kubernetes client %v", err)
+		return nil, err
+	}
+	return apiserver, err
+}
+
+func meteringToUnstructured(m *Metering) (*unstructured.Unstructured, error) {
+	values, err := ObjectToMapInterface(m)
+	if err != nil {
+		glog.Errorf("unable convert to map interface %v", err)
+		return nil, err
+	}
+	meteringDoc := &unstructured.Unstructured{}
+	meteringDoc.SetUnstructuredContent(values)
+	meteringDoc.SetKind("Sfevent")
+	meteringDoc.SetAPIVersion("instance.servicefabrik.io/v1alpha1")
+	meteringDoc.SetNamespace("default")
```

Comment:
May be you can pass as argument/object option as namespace could be changed as per use case.